src/ray/gcs/tables.cc

```diff
@@ -870,6 +870,9 @@ template class Log<JobID, ErrorTableData>;
 template class Log<ClientID, GcsNodeInfo>;
 template class Log<JobID, JobTableData>;
 template class Log<UniqueID, ProfileTableData>;
+template class Log<ClientID, HeartbeatTableData>;
```

Comment:
HeartbeatTable is inherit from Table<ClientID, HeartbeatTableData>, why add Log<ClientID, HeartbeatTableData>?

Reply:
HeartbeatTable may be inherited, but the associated Log<...> class needs to be instantiated since it is a template. If it is not then Log<ClientID, HeartbeatTableData> is not seen as a class further down resulting in the following error:
```
bazel-out/k8-opt/bin/_objs/gcs/subscription_executor.pic.o:subscription_executor.cc:function ray::gcs::SubscriptionExecutor<ray::ClientID, ray::rpc::HeartbeatTableData, ray::gcs::HeartbeatTable>::AsyncSubscribeAll(ray::ClientID const&, std::function<void (ray::ClientID const&, ray::rpc::HeartbeatTableData const&)> const&, std::function<void (ray::Status)> const&): error: undefined reference to 'ray::gcs::Log<ray::ClientID, ray::rpc::HeartbeatTableData>::Subscribe(ray::JobID const&, ray::ClientID const&, std::function<void (ray::gcs::RedisGcsClient*, ray::ClientID const&, std::vector<ray::rpc::HeartbeatTableData, std::allocator<ray::rpc::HeartbeatTableData> > const&)> const&, std::function<void (ray::gcs::RedisGcsClient*)> const&)'
collect2: error: ld returned 1 exit status
```
Just tested it with and without the line in question.